htdocs/resetbyquestions.php

```diff
@@ -132,23 +164,49 @@
     $questionValues = ldap_get_values($ldap, $entry, $answer_attribute);
     unset($questionValues["count"]);
 
-    $match = false;
+    if ($multiple_answers and $multiple_answers_one_str) {
+        # Unpack multiple questions/answers
+        $questionValues = str_getcsv($questionValues[0]);
+    }
 
-    $question = preg_quote($question,'/');
-    $answer   = preg_quote($answer,'/');
-    $pattern  = "/^\{$question\}$answer$/i";
+    if ($populate_questions) {
+        $pattern  = "/^\{(.+?)\}/i";
+        $i = 0;
+        foreach ($questionValues as $questionValue) {
+            $value = $crypt_answers ? decrypt($questionValue, $keyphrase) : $questionValue;
+            if (preg_match($pattern, $value, $matched)) {
+                $question[$i++] = $matched[1];
+            }
+            if ($i >= $questions_count) {
+                $result = "emptyresetbyquestionsform";
+                break;
+            }
+        }
+    } else {
+        # Match with user submitted values
+        $pattern  = "/^\{(.+?)\}(.+)$/i";
+        $registered_questions = [];
 
-    # Match with user submitted values
-    foreach ($questionValues as $questionValue) {
-        $value = $crypt_answers ? decrypt($questionValue, $keyphrase) : $questionValue;
-        if (preg_match($pattern, $value)) {
-            $match = true;
+        # Get registered questions
+        foreach ($questionValues as $questionValue) {
+            $value = $crypt_answers ? decrypt($questionValue, $keyphrase) : $questionValue;
+            if (preg_match($pattern, $value, $matched)) {
+                $registered_questions[$matched[1]] = $matched[2];
+            }
         }
-    }
 
-    if (!$match) {
-        $result = "answernomatch";
-        error_log("Answer does not match question for user $login");
+        $matched = 0;
+        # Match answer(s)
+        for ($q = 0; $q < $questions_count; $q++) {
+            if ($registered_questions[$question[$q]] === $answer[$q]) {
```

Comment:
hash_equals is recommended when comparing secrets
```suggestion
            if (hash_equals($registered_questions[$question[$q]], $answer[$q])) {
```